avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -189,12 +190,18 @@ private Integer schemaVersion(String topic,
                                 AvroSchema schema,
                                 Object result) {
     try {
-      Integer version;
+      Integer version = null;
       if (isDeprecatedSubjectNameStrategy(isKey)) {
         subject = getSubjectName(topic, isKey, result, schema);
       }
       AvroSchema subjectSchema = (AvroSchema) schemaRegistry.getSchemaBySubjectAndId(subject, id);
-      version = schemaRegistry.getVersion(subject, subjectSchema);
+      Metadata metadata = subjectSchema.metadata();
+      if (metadata != null) {
+        version = metadata.getConfluentVersionNumber();
+      }
+      if (version == null) {
+        version = schemaRegistry.getVersion(subject, subjectSchema);
```

Comment:
do we have such strong guarantee that `confluent:version` will always be the same as version the schema is stored under a subject?